yomigeek/mtracker-app
# #157876042 Implement User Create Request API

#### What does this PR do?
* Create an API endpoint to create a user requests

#### Description of Task to be completed?
* User should be able to create a request through this endpoint when called

#### How should this be manually tested?
* Clone the project to your local repository from git clone https://github.com/yomigeek/mtracker-app.git
* All necessary dependencies as stated in the README.md file should be installed
* run $ npm install
* Run app locally using $ npm run start
* Open the postman application and enter the URL http://localhost:5000/api/v1/users/requests with POST request method selected and also enter the required fields such as title, description and priority
```
{
	"title": "New laptop repair",
	"description":  "Laptop processor burnt",
	"priority": "medium"
}
```
* After the above is successful, you should get the following response: 
```
{
    "status": "success",
    "message": "Request created successfully!",
    "data": {
        "requests": {
            "title": "New laptop repair",
            "description": "Laptop processor burnt",
            "priority": "medium",
            "userId": 1
        }
    }
}
```
#### Any background context you want to provide?
All  fields are important i.e title, description and priority values must be supplied
Priority field value must either be low, medium or high 

####  What are the relevant pivotal tracker stories?
story type: feature
story id: 157876042

#### What are the relevant Github Issues?
None

#### Questions:
None

![request_create](https://user-images.githubusercontent.com/5827585/40546549-798ab0c0-6027-11e8-853d-04999b817ec1.PNG)
